### PR TITLE
support current_timestamp time function

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: tidb_test.go [hunk 1]
```diff
@@ -832,6 +832,21 @@ func (s *testSessionSuite) TestShow(c *C) {
 	match(c, row, "utf8_bin", "utf8", 83, "", "Yes", 1)
 }
 
+func (s *testSessionSuite) TestTimeFunc(c *C) {
+	store := newStore(c, s.dbName)
+	se := newSession(c, store, s.dbName)
+
+	last := time.Now().Format(mysql.TimeFormat)
+	r := mustExecSQL(c, se, "select now(), now(6), current_timestamp, current_timestamp(), current_timestamp(6)")
```
Comment (qiuyesuifeng) [inline tidb_test.go]: Is timestamp always equal? Seems call now() in different code line.

File: parser/parser.y [hunk 2]
```diff
@@ -2154,10 +2169,14 @@ FunctionCallNonKeyword:
...
-|	"NOW" '(' ExpressionList ')'
+|	"NOW" '(' ExpressionOpt ')'
 	{
+		args := []expression.Expression{}
+		if $3 != nil {
+			args = append(args, $3.(expression.Expression))
+		}
 		var err error
-		$$, err = expression.NewCall($1.(string), $3.([]expression.Expression),false)
+		$$, err = expression.NewCall($1.(string), args,false)
```
Comment (qiuyesuifeng) [inline parser/parser.y]: `args,false` add a space?

File: tidb_test.go (hunk 1 above)
Comment (siddontang) [inline tidb_test.go]: current_timestamp is synonymy for now, they are the same. 

btw, I don't check equal, I check greater or equal last timestamp before current execution.

**Comment by siddontang** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**